File: ChatBackend/src/main/java/ChatBackend/ChatBackend/entity/Message.java
package ChatBackend.ChatBackend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "messages")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "chat_id")
    private Chat chat;

    @ManyToOne
    @JoinColumn(name = "sender_id")
    private User sender;

    private String content;

    @Enumerated(EnumType.STRING)
    private MessageType type;

    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
    }

    public enum MessageType {
        TEXT,
        FILE,
        CALL
    }
}

File: ChatClient/data/chat/message.ts
import { Timestamp } from "react-native-reanimated/lib/typescript/reanimated2/commonTypes";

export interface Message {
  id: number;
  type: string;
  content: string;
  timestamp: string;
  fileResponses: any;
  callResponse: any;
  senderId: number;
  chatId: number;
}

export enum MessageType {
  TEXT,
  FILE,
  CALL,
}
